universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -60,11 +74,16 @@ protected ReadOnlyApplicantProgramServiceImpl(
     return getBlockAfter(block.getId());
   }
 
-  // TODO(https://github.com/seattle-uat/universal-application-tool/issues/224): Implement checking
-  // blocks for completion
   @Override
   public Optional<Block> getFirstIncompleteBlock() {
-    ImmutableList<Block> currentBlockList = getCurrentBlockList();
-    return currentBlockList.isEmpty() ? Optional.empty() : Optional.of(currentBlockList.get(0));
+    return getCurrentBlockList().isEmpty()
+        ? Optional.empty()
+        : Optional.of(getCurrentBlockList().get(0));
```

Comment:
Suggestion (blocking): as is this would the first block regardless of whether it's incomplete or not I think?

```suggestion
    : getCurrentBlockList().stream().filter(block -> !block.isCompleteWithoutErrors()).findFirst()
```